### ISSUE TITLE
EMTERPRETIFY and unary ! operator

### ISSUE THREAD

**Issue by TrevorSundberg** [kripken/emscripten#6804]
We're getting an error in js-optimizer.js where it's throwing 'yikes'. This error is almost identical to:

https://github.com/kripken/emscripten/issues/6738

I changed `throw 'yikes'` into `throw 'yikes ' + node[1]` and I got `yikes !`. Truthfully, I'm not at all familiar with what's happneing in js-optimizer or what the constant should be:

```
    case 'unary-prefix': {
      switch(node[1]) {
        case '-': return ASM_FLEXIBLE;
        case '+': return ASM_NONSIGNED; // XXX double
        case '~': return ASM_SIGNED;
        case '!': return ...; // Not sure...
        default: throw 'yikes ' + node[1];
      }
      break;
    }
```

**Comment by TrevorSundberg** [comment]
As a side note, we received this when attempting to build the entire codebase as EMTERPRETED.

**Comment by TrevorSundberg** [comment]
You're the best!

**Comment by kripken** [comment]
Heh, thanks!

Odd we never ran into this until now. Seems like it requires an expression where the sign actually matters, like `(!x) == ..`.

Fix in #6805. Although thinking about it again, I may need to change it a little.
